app/javascript/mastodon/service_worker/helpers.js

```diff
@@ -0,0 +1,167 @@
+// Various caching strategies to make entry.js easier to read. Partially
+// inspired by https://github.com/GoogleChrome/sw-toolbox
+
+const VERSION = '1.0.0'; // if necessary, we can increment this
+const CACHE_KEY = `mastodon-sw-v${VERSION}`;
+
+// taken from https://github.com/GoogleChrome/sw-toolbox/blob/master/lib/options.js
+const SUCCESS_RESPONSES = /^0|([123]\d\d)|(40[14567])|410$/
```

Comment:
I copypasta'd this from the sw-toolbox project. I checked a few different HTTP responses and it seems to correctly detect error ranges vs success ranges.